application/controllers/QuotasController.php

```diff
@@ -198,17 +189,13 @@ public function actionEditQuota($surveyid)
     {
         $surveyid = sanitize_int($surveyid);
         $oSurvey = Survey::model()->findByPk($surveyid);
-        if (!(Permission::model()->hasSurveyPermission($surveyid, 'quotas', 'update'))) {
-            Yii::app()->user->setFlash('error', gT("Access denied."));
-            $this->redirect(Yii::app()->request->urlReferrer);
-        }
-
         $quotaId = sanitize_int(Yii::app()->request->getQuery('quota_id'));
+        $this->checkPermissions($surveyid, 'update', $quotaId);
```

Comment:
If I am not wrong, usually the check permission is done using the surveyId from the quota model.

Reply:
It's a private function used here …

I can just rename it to checkPermissionAndValidity …